### PR TITLE
Sensors: Coverity fixes

### LINKED ISSUE #65328: [Coverity CID: 330682] Unchecked return value in drivers/sensor/adltc2990/adltc2990.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/adltc2990/adltc2990.c

Category: Error handling issues
Function: `adltc2990_fetch_property_value`
Component: Drivers
CID: [330682](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=330682)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/adltc2990/adltc2990.c

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #65331: [Coverity CID: 330612] Unsigned compared against 0 in drivers/sensor/tsl2561/tsl2561.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/tsl2561/tsl2561.c

Category: Control flow issues
Function: `tsl2561_sample_fetch`
Component: Drivers
CID: [330612](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=330612)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/tsl2561/tsl2561.c#L147

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_NO_EFFECT). ([CWE-570](http://cwe.mitre.org/data/definitions/570.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #65345: [Coverity CID: 330011] Unchecked return value in drivers/sensor/tsl2540/tsl2540.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/tsl2540/tsl2540.c

Category: Error handling issues
Function: `tsl2540_init`
Component: Drivers
CID: [330011](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=330011)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/tsl2540/tsl2540.c#L301

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #65346: [Coverity CID: 330000] Unchecked return value in drivers/sensor/adltc2990/adltc2990.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/adltc2990/adltc2990.c

Category: Error handling issues
Function: `adltc2990_is_busy`
Component: Drivers
CID: [330000](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=330000)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/adltc2990/adltc2990.c#L101

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #65352: [Coverity CID: 329980] Unchecked return value in drivers/sensor/tsl2540/tsl2540.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/tsl2540/tsl2540.c

Category: Error handling issues
Function: `tsl2540_attr_set`
Component: Drivers
CID: [329980](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=329980)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/tsl2540/tsl2540.c#L165

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #65374: [Coverity CID: 322697] Unchecked return value in drivers/sensor/max17055/max17055.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/max17055/max17055.c

Category: Error handling issues
Function: `max17055_write_config`
Component: Drivers
CID: [322697](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=322697)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/max17055/max17055.c#L389

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### LINKED ISSUE #65383: [Coverity CID: 322645] Unchecked return value in drivers/sensor/max17055/max17055.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/sensor/max17055/max17055.c

Category: Error handling issues
Function: `max17055_gauge_init`
Component: Drivers
CID: [322645](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=322645)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/sensor/max17055/max17055.c#L448

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_CHECKED_RETURN). ([CWE-252](http://cwe.mitre.org/data/definitions/252.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Fix of Coverity issues around sensors:

fixes #65331
fixes #65352
fixes #65346
fixes #65374
fixes #65383
fixes #65328
fixes #65345

### CODE REVIEW

File: drivers/sensor/adltc2990/adltc2990.c [hunk 1]
```diff
@@ -141,9 +141,10 @@ static int adltc2990_trigger_measurement(const struct device *dev)
 	return i2c_reg_write_byte_dt(&cfg->bus, ADLTC2990_REG_TRIGGER, 0x1);
 }
 
-static int32_t adltc2990_fetch_property_value(const struct device *dev,
+static int adltc2990_fetch_property_value(const struct device *dev,
 					      enum adltc2990_monitoring_type type,
-					      enum adltc2990_monitor_pins pin)
+					  enum adltc2990_monitor_pins pin,
+					  int32_t *output)
```
Comment (jilaypandya) [inline drivers/sensor/adltc2990/adltc2990.c:147]: Hey @benjaminbjornsson, thanks for this rework. This was actually something i was planning to do 👍 .

**Review by jilaypandya — APPROVED** [review]

**Review by nashif — DISMISSED** [review]

File: drivers/sensor/adltc2990/adltc2990.c [hunk 2]
```diff
@@ -275,15 +283,15 @@ static int adltc2990_sample_fetch(const struct device *dev, enum sensor_channel
...
+					 (float)cfg->pins_v1_v2.pins_current_resistor);
 		}
 		if (mode_v3_v4 == VOLTAGE_DIFFERENTIAL) {
 			ret = adltc2990_fetch_property_value(dev, VOLTAGE_DIFFERENTIAL, V3, &value);
 			if (ret) {
 				return ret;
 			}
 			data->pins_v3_v4_values[0] = value * (ADLTC2990_MICROOHM_CONVERSION_FACTOR /
-				 (float)cfg->pins_v3_v4.pins_current_resistor);
+					 (float)cfg->pins_v3_v4.pins_current_resistor);
```
Comment (MaureenHelm) [inline drivers/sensor/adltc2990/adltc2990.c:294]: Unrelated whitespace change

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks @benjaminbjornsson !

File: drivers/sensor/adltc2990/adltc2990.c (hunk 2 above)
Comment (benjaminbjornsson) [reply drivers/sensor/adltc2990/adltc2990.c]: Fixed.

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**